tls/s2n_x509_validator.c

```diff
@@ -457,99 +453,54 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
             break;
         }
     }
-
-    if (!issuer) {
-        goto clean_up;
-    }
+    RESULT_ENSURE(issuer != NULL, S2N_ERR_CERT_UNTRUSTED);
 
     /* Important: this checks that the stapled ocsp response CAN be verified, not that it has been verified. */
     const int ocsp_verify_res = OCSP_basic_verify(basic_response, cert_chain, validator->trust_store->trust_store, 0);
-
-    /* OCSP_basic_verify() returns 1 on success, 0 on error, or -1 on fatal error such as malloc failure. */
-    if (ocsp_verify_res != _OSSL_SUCCESS) {
-        ret_val = ocsp_verify_res == 0 ? S2N_CERT_ERR_UNTRUSTED : S2N_CERT_ERR_INTERNAL_ERROR;
-        goto clean_up;
-    }
+    RESULT_GUARD_OSSL(ocsp_verify_res, S2N_ERR_CERT_UNTRUSTED);
 
     /* do the crypto checks on the response.*/
     int status = 0;
     int reason = 0;
 
     /* sha1 is the only supported OCSP digest */
     OCSP_CERTID *cert_id = OCSP_cert_to_id(EVP_sha1(), subject, issuer);
-
-    if (!cert_id) {
-        goto clean_up;
-    }
+    RESULT_GUARD_PTR(cert_id);
```

Comment:
Same here. GUARD_PTR should only be used on functions that are defined within s2n-tls and set the errno. Otherwise prefer ENSURE.

Reply:
Right okay, that makes sense. Fixed in bf827ce6f4c671cf29d94f32af0c52c295060f34.